src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -718,24 +716,21 @@ public string GetLanguagesStringForTelemetry(Solution solution)
 
                 foreach (var projectOrDocumentId in _projectOrDocumentIds)
                 {
-                    switch (projectOrDocumentId)
+                    if (projectOrDocumentId.DocumentId is { } documentId)
```

Comment:
>is { }  [](start = 55, length = 7)

```
var documentId = projectOrDocumentId.DocumentId;
if (documentId != null)
```

would be more obvious.

Reply:
Or even better deconstruct in the `foreach`: 

```
foreach (var (projectId, documentId) in _projectOrDocumentIds)
{
   if (documentId != null)
```



---
In reply to: [362585138](https://github.com/dotnet/roslyn/pull/40670#discussion_r362585138) [](ancestors = 362585138)